drivers/hmis_csv_importer/app/jobs/hmis_csv_importer/mark_expired_job.rb

```diff
@@ -0,0 +1,55 @@
+module HmisCsvImporter
+  class MarkExpiredJob < ApplicationJob
+    def perform(now: DateTime.current, recent_imports_to_keep: 10, keep_deleted_records_for: 2.weeks)
+      @now = now.to_datetime
+      @expire_deleted_records_before = now - keep_deleted_records_for
+      @recent_imports_to_keep = recent_imports_to_keep
+
+      total = 0
+      models.each do |model|
+        # FIXME, should include select clause to improve performance
+        candidates_for_expiration = model.
+          where.not(loader_id: recent_import_ids).
+          preload(:destination_record_with_deleted)
```

Comment:
This probably isn't quite right, but it should be close.
```suggestion
          preload(:destination_record_with_deleted).
          select(:id, model.hud_key, data_source_id, :DateDeleted, :loader_id)
```